### DISCUSSION TITLE
How to add trace_id to logger metadata

### DISCUSSION THREAD

**Discussion by btkostner** [open-telemetry/opentelemetry-erlang#391] [Q&A]
For some background, I am currently using opentelemetry with Elixir and exported to datadog via dd-agent. It's working great and everything tracing wise _just works_ :tm:. All traces show up in datadog with an `otel.trace_id` attribute, but datadog expects log messages to also have that `trace_id` to correlate between the two. So my question is, what is the best way to add the `trace_id` in `Logger.metadata`?

**Comment by tsloughter** [comment]
I'm working on a useful response. I first have to get all the lost context from the discussions we've had about this back into my head because I can't find where it was written down :)

I think we decided on adding functionality in otel that updates the logger process metadata when a span is started with the tracer macros.

But there may also be a way to do it with `logger_filters` that wouldn't require updates to opentelemetry.

Will get back to you later today.

**Comment by tsloughter** [comment]
Oh, @bryannaegele just pointed me to most of the existing context of our discussions! https://github.com/salemove/opentelemetry_logger_metadata/issues/2#issuecomment-1084933387

This also points to a way you can easily do it today by just using that application since you are using Elixir.

Looks like from my last message there I was pushing for not going the filter or OnStart processor route and instead adding it when the macro to start an active span is called. I should be able to get a PR for that done today or tomorrow. I think the biggest question on that will be if we need to have an option to enable/disable it.

**Reply by bryannaegele** [reply]
I think we need to also handle it in context attach/detach and not just the macros.

**Reply by tsloughter** [reply]
Ah right right

**Reply by tsloughter** [reply]
Looks like the only place it needs to be added is in `attach` and `detach` since that is what gets used by `with_span`.

**Comment by tsloughter** [comment]
Opened a PR https://github.com/open-telemetry/opentelemetry-erlang/pull/394
